src/tools/compiletest/src/util.rs

```diff
@@ -106,7 +106,7 @@ macro_rules! static_regex {
 pub(crate) use static_regex;
 
 macro_rules! string_enum {
-    ($(#[$meta:meta])* $vis:vis enum $name:ident { $($variant:ident => $repr:expr,)* }) => {
+    (no_from_str, $(#[$meta:meta])* $vis:vis enum $name:ident { $($variant:ident => $repr:expr,)* }) => {
```

Comment:
Remark: I'm not a huge fan of `string_enum` or adding even more complexity to it, but ditto on just spelling the actual test suites out.